lib/libc/Kconfig

```diff
@@ -145,6 +145,20 @@ config MINIMAL_LIBC_OPTIMIZE_STRING_FOR_SIZE
 	  Enable smaller but potentially slower implementations of memcpy and
 	  memset. On the Cortex-M0+ this reduces the total code size by 120 bytes.
 
+config MINIMAL_LIBC_RAND
+	bool "Enables rand and srand functions"
+	default y
```

Comment:
Should not be `default y` as per https://github.com/zephyrproject-rtos/zephyr/pull/32667#discussion_r656414206.

Any **minimal** C library features that predicate on such a heavy requirement as requiring an additional memory partition should not be enabled by default, and should only be enabled on demand.

Reply:
done.  Thanks.